pkg/component/component.go

```diff
@@ -402,6 +340,203 @@ func getS2IPaths(podEnvs []corev1.EnvVar) []string {
 	return retVal
 }
 
+// CreateComponent creates component as per the passed component settings
+//	Parameters:
+//		client: occlient instance
+//		componentConfig: the component configuration that holds all details of component
+//		context: the component context indicating the location of component config and hence its source as well
+//		stdout: io.Writer instance to write output to
+//	Returns:
+//		err: errors if any
+func CreateComponent(client *occlient.Client, componentConfig config.LocalConfigInfo, context string, stdout io.Writer) (err error) {
+
+	cmpName := componentConfig.GetName()
+	cmpType := componentConfig.GetType()
+	cmpSrcType := componentConfig.GetSourceType()
+	cmpPorts := componentConfig.GetPorts()
+	cmpSrcRef := componentConfig.GetRef()
+	appName := componentConfig.GetApplication()
```

Comment:
It is already being pulled from a already populated struct but its making func calls only to avoid spreading default values of the structs to many places